app/src/main/java/de/danoeh/antennapod/dialog/EpisodesApplyActionFragment.java

```diff
@@ -0,0 +1,414 @@
+                    return code;
+                }
+            }
+        });
+        refreshTitles();
+        refreshCheckboxes();
+    }
+
+    private void sortByDuration(final boolean reverse) {
+        Collections.sort(episodes, new Comparator<FeedItem>() {
+            @Override
+            public int compare(FeedItem lhs, FeedItem rhs) {
+                int code = lhs.getMedia().getDuration() - rhs.getMedia().getDuration();
+                if (reverse) {
+                    return -1 * code;
+                } else {
+                    return code;
+                }
+            }
+        });
+        refreshTitles();
+        refreshCheckboxes();
+    }
+
+    private void checkAll() {
+        for (FeedItem episode : episodes) {
+            if(false == checkedIds.contains(episode.getId())) {
+                checkedIds.add(episode.getId());
+            }
+        }
+        refreshCheckboxes();
+    }
+
+    private void checkNone() {
+        checkedIds.clear();
+        refreshCheckboxes();
+    }
+
+    private void checkPlayed(boolean isPlayed) {
+        for (FeedItem episode : episodes) {
+            if(episode.isRead() == isPlayed) {
+                if(!checkedIds.contains(episode.getId())) {
+                    checkedIds.add(episode.getId());
+                }
+            } else {
+                if(checkedIds.contains(episode.getId())) {
+                    checkedIds.remove(episode.getId());
+                }
+            }
+        }
+        refreshCheckboxes();
+    }
+
+    private void checkDownloaded(boolean isDownloaded) {
+        for (FeedItem episode : episodes) {
+            if(episode.getMedia().isDownloaded() == isDownloaded) {
+                if(!checkedIds.contains(episode.getId())) {
+                    checkedIds.add(episode.getId());
+                }
+            } else {
+                if(checkedIds.contains(episode.getId())) {
+                    checkedIds.remove(episode.getId());
+                }
+            }
+        }
+        refreshCheckboxes();
+    }
+
+    private void refreshTitles() {
+        titles.clear();
+        for(FeedItem episode : episodes) {
+            titles.add(episode.getTitle());
+        }
+        mAdapter.notifyDataSetChanged();
+    }
+
+    private void refreshCheckboxes() {
+        for (int i = 0; i < episodes.size(); i++) {
+            FeedItem episode = episodes.get(i);
+            boolean checked = checkedIds.contains(episode.getId());
+            mListView.setItemChecked(i, checked);
+        }
+        ActivityCompat.invalidateOptionsMenu(EpisodesApplyActionFragment.this.getActivity());
+    }
+
+    private void queueChecked() {
+        LongList orderedIds = new LongList();
+        for(FeedItem episode : episodes) {
+            if(checkedIds.contains(episode.getId())) {
+               orderedIds.add((episode.getId()));
+            }
+        }
+        DBWriter.addQueueItem(getActivity(), false, orderedIds.toArray());
```

Comment:
for people that have 'enqueueAtFront()' set this will wind up giving them the opposite of what they expect. Do we think that's OK?  Should we try to fix that?